src/container-utils.ts

```diff
@@ -123,3 +126,68 @@ export async function removeContainerIfExists(engineId: string, container: strin
     throw new Error('There was an error removing the container: ' + e);
   }
 }
+
+// Get all volume names that match the name of the container
+async function getVolumesMatchingContainer(engineId: string, container: string): Promise<string[]> {
+  try {
+    // Volumes returns a list of volumes across all engines. Only get the volume that matches our engineId
+    const engineVolumes = await extensionApi.containerEngine.listVolumes();
+    if (!engineVolumes) {
+      throw new Error('No volume providers found');
+    }
+    // If none are found, just return an empty array / small warning
+    const volumes = engineVolumes.find(volume => volume.engineId === engineId);
+    if (!volumes) {
+      console.log('Ignoring removing volumes, no volume engine providers found for engineId: ', engineId);
+      return [];
+    }
+
+    // Go through each volume and only retrieve the ones that match our container
+    // "Names" in the API weirdly has `/` appended to the beginning of the name due to how it models the podman API
+    // so we have to make sure / is appended to the container name for comparison..
+    let volumeNames = [];
+    volumes.Volumes.forEach(v => {
+      v.containersUsage.forEach(c => {
+        c.names.forEach(n => {
+          if (n === '/' + container) {
+            volumeNames.push(v.Name);
+          }
+        });
+      });
+    });
+
+    // Remove any duplicates that may have been added (same volume used multiple times / mounted).
+    volumeNames = [...new Set(volumeNames)];
+    return volumeNames;
+  } catch (e) {
+    console.log(e);
+    throw new Error('There was an error getting the volumes: ' + e);
```

Comment:
If we have an error getting volumes this means we wouldn't even try to remove the container. IMHO we should still try, i.e. either don't throw this or put a catch around line 174.

Reply:
I've updated the function so that even if we error getting the volumes, we'll still try and remove the container.